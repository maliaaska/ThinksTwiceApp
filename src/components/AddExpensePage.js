import React from 'react';
import {BrowserRouter, Route, Switch, Link, NavLink } from 'react-router-dom';

const AddExpensePage = () => (
  <div>
  This is from Create component
  </div>
);

export default AddExpensePage;