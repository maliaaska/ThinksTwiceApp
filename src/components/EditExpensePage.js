import React from 'react';

const EditExpensePage = (props) => {
  console.log(props);
  return (
  <div>
 Editing expense with ID of {props.match.params.id}
  </div>
);
}
export default EditExpensePage;