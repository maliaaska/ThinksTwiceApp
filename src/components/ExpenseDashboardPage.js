import React from 'react';
import {BrowserRouter, Route, Switch, Link, NavLink } from 'react-router-dom';


const ExpenseDashboardPage = () => (
  <div>
  This is from dashboard page component
  </div>
)

export default ExpenseDashboardPage;