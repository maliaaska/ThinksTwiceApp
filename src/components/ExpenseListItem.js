import React from 'react';
import { connect } from 'react-redux';
import { removeExpense } from '../actions/expenses';
import { Link } from 'react-router-dom';
import moment from 'moment';
import numeral from 'numeral';

export const ExpenseListItem = ({dispatch, description, amount, createdAt, id}) => (
  <div>
    <Link to={`/edit/${id}`}>
      <h3>{description}</h3>
    </Link>   
    <p>
      {numeral(amount / 100).format('$0,0.00')} 
      - 
      {moment(createdAt).format('Do MMMM, YYYY')}
    </p>
     
  </div>
);

export default connect()(ExpenseListItem);
