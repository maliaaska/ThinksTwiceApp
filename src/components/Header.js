import React from 'react';
import { NavLink } from 'react-router-dom';

const Header = () => (
  <header>
    <h1>ThinkTwice</h1>
    <NavLink to="/" activeClassName="is-active" exact={true}>Dashboard </NavLink>
    <NavLink to="/create" activeClassName="is-active">Expense Create </NavLink>
  </header>
);

export default Header;
