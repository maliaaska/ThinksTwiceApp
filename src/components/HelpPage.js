import React from 'react';
import {BrowserRouter, Route, Switch, Link, NavLink } from 'react-router-dom';

const HelpPage = () => (
  <div>
  This is from Help component
  </div>
);

export default HelpPage;