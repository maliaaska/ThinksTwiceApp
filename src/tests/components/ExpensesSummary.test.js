import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';

test('should corectly render expense summary with one expense', () => {
  const wrapper = shallow(<ExpensesSummary expenseCount={1} expensesTotal={235} />)
  expect(wrapper).toMatchSnapshot();
});

test('should corectly render expense summary with multiple expense', () => {
  const wrapper = shallow(<ExpensesSummary expenseCount={23} expensesTotal={234567653345} />)
  expect(wrapper).toMatchSnapshot();
});