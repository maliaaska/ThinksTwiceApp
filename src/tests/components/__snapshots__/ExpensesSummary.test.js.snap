// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should corectly render expense summary with multiple expense 1`] = `
<div>
  <h1>
    Viewing 
    23
     
    expenses
     totalling 
    $2,345,676,533.45
     
  </h1>
</div>
`;

exports[`should corectly render expense summary with one expense 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    expense
     totalling 
    $2.35
     
  </h1>
</div>
`;
